src/plugins/orangehrmLDAPAuthenticationPlugin/entity/LDAPSyncStatus.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Entity;
+
+use DateTime;
+use Doctrine\ORM\Mapping as ORM;
+use OrangeHRM\Entity\Decorator\DecoratorTrait;
+use OrangeHRM\Entity\Decorator\LDAPSyncStatusDecorator;
+
+/**
+ * @method LDAPSyncStatusDecorator getDecorator()
+ *
+ * @ORM\Table(name="ohrm_ldap_sync_status")
+ * @ORM\Entity
+ */
+class LDAPSyncStatus
+{
+    use DecoratorTrait;
+
+    public const SYNC_STATUS_FAILED = 0;
+    public const SYNC_STATUS_SUCCEEDED = 1;
+    public const SYNC_STATUS_NOT_AVAILABLE = 2;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private int $id;
+
+    /**
+     * @var DateTime|null
+     *
+     * @ORM\Column(name="sync_started_at", type="datetime", length=25, nullable=false)
+     */
+    private ?DateTime $syncStartedAt = null;
+
+    /**
+     * @var DateTime|null
+     *
+     * @ORM\Column(name="sync_finished_at", type="datetime", length=25, nullable=true)
+     */
+    private ?DateTime $syncFinishedAt = null;
+
+    /**
+     * @var User|null
+     * @ORM\ManyToOne(targetEntity="OrangeHRM\Entity\User")
+     * @ORM\JoinColumn(name="synced_by", referencedColumnName="id")
+     */
+    private ?User $syncedBy = null;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(name="sync_status", type="integer")
+     */
+    private int $syncStatus;
```

Comment:
```suggestion
    private int $syncStatus = self::SYNC_STATUS_FAILED;
```
not necessary. but can define default status if you want